### DISCUSSION TITLE
Rich text widget

### DISCUSSION THREAD

**Discussion by White-Rabbit-Scientific** [beeware/toga#1692] [Q&A]
Self explanatory: is there a rich text widget?

I would like a way of displaying text for live updates, kind of like how G-code is displayed line by line as it is passed to a 3D printer.

Whatever the solution, will it be able to handle hundreds (or even thousands) of lines of text, with new text being appended to the end?

Best options I can see are: using a label inside a scroll window, or embedding a web browser window and pointing the browser to a local HTML file and having it refresh at regular time intervals.

**Comment by freakboy3742** [comment]
No, there isn't but adding one is definitely on the longer term wish list.

As you note, a short term workaround is a web-based widget. Using local HTML isn't a great option because of security limitations on web widgets; you'll find it a lot easier to run a web server in a background thread, and use that to serve content. That approach also gives you a lot more flexibility about serving the JS and image resources needed to support the HTML.
